clippy_lints/src/item_name_repetitions.rs

```diff
@@ -180,7 +181,8 @@ fn have_no_extra_prefix(prefixes: &[&str]) -> bool {
 }
 
 fn check_fields(cx: &LateContext<'_>, threshold: u64, item: &Item<'_>, fields: &[FieldDef<'_>]) {
-    if (fields.len() as u64) < threshold {
+    if (fields.len() as u64) < threshold || any_impl_has_lint_allowed(cx, STRUCT_FIELD_NAMES, item.owner_id.to_def_id())
```

Comment:
thought: unsure if we should perform this check before or after the field name check

this is a bunch of queries, that is some string comparisons, so probably before makes sense.